### DISCUSSION TITLE
Does Visidata have cell formulas?

### DISCUSSION THREAD

**Discussion by halloleo** [saulpw/visidata#1588] [Q&A]
I use Visidata more and more, but there's still a use case I reach for Excel: When I quickly want to model something with some _cell_ formulas.

I know I can _apply_ transformations to columns, create new columns via external formulas from given columns, etc in Visidata. But In Excel it is _very_ powerful to have a formula _in a cell_ which then auto-updates.

Does Visidata have this and I just missed it?

***

Side note: I started to look at `sc-im` for this, but I'm a lot more familiar with `vd` and it's powers, so if I find this here, I'd stick with Visidata. 😄

**Comment by saulpw** [comment]
Well, what if there were a new type of Sheet, which had `varname`, `formula`, and `eval(formula)` columns, and you could use the values in the `varname` column in the `formula` column to refer to the eval'ed result.  How far does that get you?

**Reply by halloleo** [reply]
Interesting! Definitely something to try out!

Do I understand it correctly that you suggest something like  a 1-dimensional "Excel"? -- The advantage over Excel would be that the flow is probably easier to follow; the disadvantages might be that it needs more thought to organise your model.

**Reply by saulpw** [reply]
Yes, exactly, a 1D Excel.  It might require thinking a bit differently than we're used to with cell-based spreadsheets, but of course VisiData is not a spreadsheet, with all the incumbent advantages and disadvantages =)

The nice thing about doing it this way, is that it would use the same core lookup functionality that some other people want (e.g. #1648), and otherwise just take an hour or two of putting it together.

**Reply by halloleo** [reply]
Cool! Thanks for confirming this.
